IaC/main.tf

```diff
@@ -1,433 +1,32 @@
-  api_id    = aws_apigatewayv2_api.CatalogAPIGW.id
-  route_key = "$default"
-
-  target = "integrations/${aws_apigatewayv2_integration.CatalogAPIGWIntegration.id}"
-}
-
-resource "aws_lambda_permission" "CatalogAPIGWPermission" {
-  statement_id  = "CatalogAPIGWPermission"
-  action        = "lambda:InvokeFunction"
-  function_name = aws_lambda_function.CatalogAPILambda.function_name
-  principal     = "apigateway.amazonaws.com"
-
-  source_arn = "${aws_apigatewayv2_api.CatalogAPIGW.execution_arn}/*/*"
-}
-
-resource "aws_sns_topic" "CartAPITopic" {
-  name = "CartAPITopic"
-}
-
-resource "aws_sqs_queue" "OrdersAPICartsQueue" {
-  name = "OrdersAPICartsQueue"
-}
-
-resource "aws_sqs_queue_policy" "OrdersAPICartsQueuePolicy" {
-  queue_url = aws_sqs_queue.OrdersAPICartsQueue.id
-
-  policy = <<POLICY
-{
-  "Version": "2012-10-17",
-  "Id": "sqspolicy",
-  "Statement": [
-    {
-      "Sid": "First",
-      "Effect": "Allow",
-      "Principal": "*",
-      "Action": "sqs:SendMessage",
-      "Resource": "${aws_sqs_queue.OrdersAPICartsQueue.arn}",
-      "Condition": {
-        "ArnEquals": {
-          "aws:SourceArn": "${aws_sns_topic.CartAPITopic.arn}"
-        }
-      }
-    }
-  ]
-}
-POLICY
-}
-
-resource "aws_sns_topic_subscription" "OrdersAPICartsQueueSubscription" {
-  topic_arn = aws_sns_topic.CartAPITopic.arn
-  protocol  = "sqs"
-  endpoint  = aws_sqs_queue.OrdersAPICartsQueue.arn
-}
-
-resource "aws_sns_topic" "OdersAPITopic" {
-  name = "OdersAPITopic"
-}
-
-resource "aws_sqs_queue" "CatalogAPIOrdersQueue" {
-  name = "CatalogAPIOrdersQueue"  
-}
-
-resource "aws_sqs_queue_policy" "CatalogAPIOrdersQueuePolicy" {
-  queue_url = aws_sqs_queue.CatalogAPIOrdersQueue.id
-
-  policy = <<POLICY
-{
-  "Version": "2012-10-17",
-  "Id": "sqspolicy",
-  "Statement": [
-    {
-      "Sid": "First",
-      "Effect": "Allow",
-      "Principal": "*",
-      "Action": "sqs:SendMessage",
-      "Resource": "${aws_sqs_queue.CatalogAPIOrdersQueue.arn}",
-      "Condition": {
-        "ArnEquals": {
-          "aws:SourceArn": "${aws_sns_topic.OdersAPITopic.arn}"
-        }
-      }
-    }
-  ]
-}
-POLICY
-}
-
-resource "aws_sns_topic_subscription" "CatalogAPIOrdersQueueSubscription" {
-  topic_arn = aws_sns_topic.OdersAPITopic.arn
-  protocol  = "sqs"
-  endpoint  = aws_sqs_queue.CatalogAPIOrdersQueue.arn
-}
-
-
-
-output "DBHost" {
-  value = aws_db_instance.MTBZoneDB.address
-}
-
-output "CatalogAPIUrl" {
-  value = aws_apigatewayv2_stage.CatalogAPIGWStage.invoke_url
-}
-
-output "CatalogAPIOrdersQueue" {
-  value = aws_sqs_queue.CatalogAPIOrdersQueue.url
-}
-
-output "OdersAPITopic" {
-  value = aws_sns_topic.OdersAPITopic.arn
-}
-
-output "CartAPITopic" {
-  value = aws_sns_topic.CartAPITopic.arn
-}
-
-output "OrdersAPICartsQueue" {
-  value = aws_sqs_queue.OrdersAPICartsQueue.url
+module "CatalogAPILambda"{
+    source = "lamda_module_template.tf" // sau "", pt ca e in acelasi folder?
```

Comment:
* Module needs to be in a dedicated folder
* Here you need to reference that folder, i.e. `./lambda_api_module`